### PR TITLE
[Android] Don't dispose the _labelTextColorDefault on Label Fast Renderer

### LINKED ISSUE #6994: [Bug] Regression in Xamarin.Forms 4.2.0-pre1 (Java.Lang.NullPointerException when using FastRenderers)
### Description

My android app is crashing when using XF 4.2.0-pre1. 
No exception with XF 4.1.0, so I think this is a regression.
Also, no exception with XF 4.2.0-pre1 if I use LegacyRenderers instead of FastRenderers.

### Steps to Reproduce

1. Open the attached solution, build and launch the app
2. Click any item in the list on the left

### Expected Behavior

The clicked item should be selected (and its color should change).

### Actual Behavior

App crashes with the following exception:

```
07-27 19:42:53.567 I/MonoDroid(29000): Java.Lang.NullPointerException: Exception of type 'Java.Lang.NullPointerException' was thrown.
07-27 19:42:53.567 I/MonoDroid(29000):   at Java.Interop.JniEnvironment+InstanceMethods.CallNonvirtualVoidMethod (Java.Interop.JniObjectReference instance, Java.Interop.JniObjectReference type, Java.Interop.JniMethodInfo method, Java.Interop.JniArgumentValue* args) [0x00089] in <d62a515dd509498b9888afef1e7ee082>:0 
07-27 19:42:53.567 I/MonoDroid(29000):   at Java.Interop.JniPeerMembers+JniInstanceMethods.InvokeVirtualVoidMethod (System.String encodedMember, Java.Interop.IJavaPeerable self, Java.Interop.JniArgumentValue* parameters) [0x0005d] in <d62a515dd509498b9888afef1e7ee082>:0 
07-27 19:42:53.567 I/MonoDroid(29000):   at Android.Widget.TextView.SetTextColor (Android.Content.Res.ColorStateList colors) [0x00031] in <91096acba71b4cd4bc97355af9ac8959>:0 
07-27 19:42:53.567 I/MonoDroid(29000):   at Xamarin.Forms.Platform.Android.FastRenderers.LabelRenderer.UpdateColor () [0x0002b] in D:\a\1\s\Xamarin.Forms.Platform.Android\FastRenderers\LabelRenderer.cs:277 
07-27 19:42:53.567 I/MonoDroid(29000):   at Xamarin.Forms.Platform.Android.FastRenderers.LabelRenderer.UpdateText () [0x00092] in D:\a\1\s\Xamarin.Forms.Platform.Android\FastRenderers\LabelRenderer.cs:359 
07-27 19:42:53.567 I/MonoDroid(29000):   at Xamarin.Forms.Platform.Android.FastRenderers.LabelRenderer.OnElementPropertyChanged (System.Object sender, System.ComponentModel.PropertyChangedEventArgs e) [0x0005f] in D:\a\1\s\Xamarin.Forms.Platform.Android\FastRenderers\LabelRenderer.cs:254 
07-27 19:42:53.567 I/MonoDroid(29000):   at (wrapper delegate-invoke) <Module>.invoke_void_object_PropertyChangedEventArgs(object,System.ComponentModel.PropertyChangedEventArgs)
07-27 19:42:53.567 I/MonoDroid(29000):   at Xamarin.Forms.BindableObject.OnPropertyChanged (System.String propertyName) [0x00000] in D:\a\1\s\Xamarin.Forms.Core\BindableObject.cs:211 
07-27 19:42:53.567 I/MonoDroid(29000):   at Xamarin.Forms.Element.OnPropertyChanged (System.String propertyName) [0x00000] in D:\a\1\s\Xamarin.Forms.Core\Element.cs:359 
07-27 19:42:53.567 I/MonoDroid(29000):   at Xamarin.Forms.BindableObject.ClearValue (Xamarin.Forms.BindableProperty property, System.Boolean fromStyle, System.Boolean checkAccess) [0x000c6] in D:\a\1\s\Xamarin.Forms.Core\BindableObject.cs:78 
07-27 19:42:53.567 I/MonoDroid(29000):   at Xamarin.Forms.BindableObject.ClearValue (Xamarin.Forms.BindableProperty property) [0x00000] in D:\a\1\s\Xamarin.Forms.Core\BindableObject.cs:30 
07-27 19:42:53.568 I/MonoDroid(29000):   at Xamarin.Forms.Setter.UnApply (Xamarin.Forms.BindableObject target, System.Boolean fromStyle) [0x00084] in D:\a\1\s\Xamarin.Forms.Core\Setter.cs:102 
07-27 19:42:53.568 I/MonoDroid(29000):   at Xamarin.Forms.TriggerBase.OnConditionChanged (Xamarin.Forms.BindableObject bindable, System.Boolean oldValue, System.Boolean newValue) [0x00074] in D:\a\1\s\Xamarin.Forms.Core\Interactivity\TriggerBase.cs:106 
07-27 19:42:53.568 I/MonoDroid(29000):   at Xamarin.Forms.BindingCondition.OnBoundPropertyChanged (Xamarin.Forms.BindableObject bindable, System.Object oldValue, System.Object newValue) [0x0001d] in D:\a\1\s\Xamarin.Forms.Core\Interactivity\BindingCondition.cs:95 
07-27 19:42:53.568 I/MonoDroid(29000):   at Xamarin.Forms.BindableObject.SetValueActual (Xamarin.Forms.BindableProperty property, Xamarin.Forms.BindableObject+BindablePropertyContext context, System.Object value, System.Boolean currentlyApplying, Xamarin.Forms.Internals.SetValueFlags attributes, System.Boolean silent) [0x00120] in D:\a\1\s\Xamarin.Forms.Core\BindableObject.cs:445 
07-27 19:42:53.568 I/MonoDroid(29000):   at Xamarin.Forms.BindableObject.SetValueCore (Xamarin.Forms.BindableProperty property, System.Object value, Xamarin.Forms.Internals.SetValueFlags attributes, Xamarin.Forms.BindableObject+SetValuePrivateFlags privateAttributes) [0x00173] in D:\a\1\s\Xamarin.Forms.Core\BindableObject.cs:379 
07-27 19:42:53.568 I/MonoDroid(29000):   at Xamarin.Forms.BindingExpression.ApplyCore (System.Object sourceObject, Xamarin.Forms.BindableObject target, Xamarin.Forms.BindableProperty property, System.Boolean fromTarget) [0x00220] in D:\a\1\s\Xamarin.Forms.Core\BindingExpression.cs:167 
07-27 19:42:53.568 I/MonoDroid(29000):   at Xamarin.Forms.BindingExpression.Apply (System.Boolean fromTarget) [0x0003e] in D:\a\1\s\Xamarin.Forms.Core\BindingExpression.cs:57 
07-27 19:42:53.568 I/MonoDroid(29000):   at Xamarin.Forms.BindingExpression+BindingExpressionPart.<PropertyChanged>b__49_0 () [0x00000] in D:\a\1\s\Xamarin.Forms.Core\BindingExpression.cs:618 
07-27 19:42:53.568 I/MonoDroid(29000):   at Java.Lang.Thread+RunnableImplementor.Run () [0x00008] in <91096acba71b4cd4bc97355af9ac8959>:0 
07-27 19:42:53.568 I/MonoDroid(29000):   at Java.Lang.IRunnableInvoker.n_Run (System.IntPtr jnienv, System.IntPtr native__this) [0x00009] in <91096acba71b4cd4bc97355af9ac8959>:0 
07-27 19:42:53.568 I/MonoDroid(29000):   at (wrapper dynamic-method) Android.Runtime.DynamicMethodNameCounter.27(intptr,intptr)
07-27 19:42:53.568 I/MonoDroid(29000):   --- End of managed Java.Lang.NullPointerException stack trace ---
07-27 19:42:53.568 I/MonoDroid(29000): java.lang.NullPointerException
07-27 19:42:53.568 I/MonoDroid(29000): 	at android.widget.TextView.setTextColor(TextView.java:4023)
07-27 19:42:53.568 I/MonoDroid(29000): 	at mono.java.lang.RunnableImplementor.n_run(Native Method)
07-27 19:42:53.568 I/MonoDroid(29000): 	at mono.java.lang.RunnableImplementor.run(RunnableImplementor.java:30)
07-27 19:42:53.568 I/MonoDroid(29000): 	at android.os.Handler.handleCallback(Handler.java:790)
07-27 19:42:53.568 I/MonoDroid(29000): 	at android.os.Handler.dispatchMessage(Handler.java:99)
07-27 19:42:53.568 I/MonoDroid(29000): 	at android.os.Looper.loop(Looper.java:164)
07-27 19:42:53.568 I/MonoDroid(29000): 	at android.app.ActivityThread.main(ActivityThread.java:6494)
07-27 19:42:53.568 I/MonoDroid(29000): 	at java.lang.reflect.Method.invoke(Native Method)
07-27 19:42:53.568 I/MonoDroid(29000): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:438)
07-27 19:42:53.568 I/MonoDroid(29000): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:807)
```

### Basic Information

- Version with issue: XF 4.2.0-pre1 (or pre2)
- Last known good version: XF 4.1.0.618606
- IDE: VS 16.3.0 preview 1.0
- Platform Target Frameworks: Android 9.0

### Reproduction Link

[TestXF4.2.zip](https://github.com/xamarin/Xamarin.Forms/files/3438436/TestXF4.2.zip)

### PR DESCRIPTION
**Pull request by samhouts**
### Description of Change ###

#6467 disposes the `_labelTextColorDefault` on the fast LabelRenderer for Android. This causes a NullPointerException when attempting to access it later.

This does not happen on the legacy renderer, which now also has the same dispose but a slightly different pattern of setting the value. Using the same pattern does not fix the crash on the Fast Renderer, so the best thing is to remove the dispose and possibly revisit this later.

### Issues Resolved ### 

- fixes #6994 

### API Changes ###

 
 None

### Platforms Affected ### 

- Android

### Before/After Screenshots ### 
Not applicable

### Testing Procedure ###

There is an automated test with instructions.

### PR Checklist ###

- [ ] Has automated tests 
- [ ] Rebased on top of the target branch at time of PR
- [ ] Changes adhere to coding standard

### CODE REVIEW

File: Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue6994.cs [hunk 1]
```diff
@@ -74,7 +74,9 @@ protected override void Init()
 						await Task.Delay(20);
 						source.Clear();
 						await Task.Delay(2000);
+#pragma warning disable CS4014
 						Device.InvokeOnMainThreadAsync(() => instructions.Text = "Success");
```
Comment (jfversluis) [inline Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue6994.cs]: I think this was intentional, so I disabled the warning, if not, await it :) @samhouts

**Comment by hartez** [comment]
The old-style renderer shouldn't be disposing of this value, either, but we're getting away with it because of the container.

> possibly revisit this later
I don't think we need to; it's unlikely that an extra reference to a member of the same class is causing a leak.

**Comment by kvpt** [comment]
@samhouts I tested your reproduction, thanks for doing that, I confirm that removing the dispose labelTextColorDefault fix the issue.
But I think this reveal that something is broken somewhere else.
The only way, for me, that the dispose can cause this issue is a race condition between dispose method and OnElementPropertyChanged event handler. Now that the unsubscribing of the handler is at the beginning of the method the probability that this occur is greatly reduced.
To confirm that it's not the issue here, I used a ManualResetEventSlim to prevent that, it not fix the crash so it's confirm that it's not the cause.

So I made some more testing after that, and found that removing the Trigger make the crash disappear.
Other thing really weird, you can test it easily, on your test case, the label dispose (so the method fixed here) is only called once, and on the top label.
So if the crash happen, it's implied that the UpdateColor is called on the top label and this label doesn't have trigger either, so how removing them fix the issue is really weird.
Other question, why the top label is disposed in the first place ? at this time, it make no sense.

From the stacktrace it's cause by :
https://github.com/xamarin/Xamarin.Forms/blob/86650d419f05fff4aff51d239ed3b62acf27d514/Xamarin.Forms.Platform.Android/VisualElementPackager.cs#L269
If I remove it no more crash, and if I navigate back the dispose is called, so I don't know if it a right thing or not.
Is the label rendered need to be destroyed and recreated if you do that ?
```cs
stackLayout.Children.Remove(instructions);
stackLayout.Children.Insert(0, instructions);
```
Actually it's the case.

It will need some more in depth testing.

**Comment by samhouts** [comment]
@kvpt @hartez I agree with both of your assessments. This was a strange crash to me. Thanks!

**Comment by samhouts** [comment]
Failing tests don't look related. Also can't repro locally.

**Comment by PureWeen** [comment]
Yea the Android failure is one I see randomly with Android tests :-/ It seems like the UI test runner just crashes for some reason. I have an issue logged here for it https://github.com/microsoft/appcenter/issues/893

Similarly with the CollectionView failure on iOS for some reason it seems to fail to navigate to the test sometimes and it just gets stuck on the screen listing the tests

**Pull request merged**